api/mutations/communityMember/addCommunityMembers.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+import type { GraphQLContext } from '../../';
+import type { DBCommunity } from 'shared/types';
+import { getCommunityById } from '../../models/community';
+import {
+  createMemberInCommunity,
+  checkUserPermissionsInCommunity,
+} from '../../models/usersCommunities';
+import { createMemberInDefaultChannels } from '../../models/usersChannels';
+import { isAuthedResolver as requireAuth } from '../../utils/permissions';
+
+type Input = {
+  input: {
+    communityIds: Array<string>,
+  },
+};
+
+export default requireAuth(async (_: any, args: Input, ctx: GraphQLContext) => {
+  const { user } = ctx;
+  const { communityIds } = args.input;
+
+  const canJoinCommunity = async (
+    communityId: string
+  ): Promise<?DBCommunity> => {
+    const [permissions, community] = await Promise.all([
+      checkUserPermissionsInCommunity(communityId, user.id),
+      getCommunityById(communityId),
+    ]);
+
+    if (!community) {
+      return null;
+    }
+
+    // if no permissions exist, join them to the community!
+    if (!permissions || permissions.length === 0) {
+      return community;
+    }
+
+    const permission = permissions[0];
+
+    if (!permission.isMember) {
+      return community;
+    }
+
+    return null;
+  };
+
+  const handleJoin = async (communityId: string) => {
+    return await Promise.all([
+      createMemberInCommunity(communityId, user.id),
+      createMemberInDefaultChannels(communityId, user.id),
+    ]);
+  };
+
+  return communityIds.map(async id => {
```

Comment:
This is kinda awkward since it requires a lot of db queries which could be intensive. Can we just check perms and join all the communities with the plural db queries so we only do a single one?

Reply:
Do you have some ideas on how this could be structured? I'm not sure I see a clear way around having to check each passed community id and user id for permissions. If we do them in bulk, we'd have to somehow account for `null` results, right?

Anyways - I definitely agree this *could* be intensive if it's like 20+ communities being joined. We can brainstorm on this together if that's cool.

Reply:
Not sure, haven't spent a ton of time thinking about it. :blush: If not this is fine too, but then let's limit the amount you can join in one go so people don't spam-join 10000 communities at once!

Reply:
Added a limit of 20 for now :)